include/aspect/material_model/interface.h

```diff
@@ -928,6 +928,42 @@ namespace aspect
     compute_volume_fractions(const std::vector<double> &compositional_fields,
                              const ComponentMask &field_mask = ComponentMask());
 
+
+    /**
+     * For multicomponent material models:
+     * Enumeration for selecting which averaging scheme to use when
+     * averaging the properties of different compositional fields.
+     * Select between harmonic, arithmetic, geometric, and
+     * maximum_composition. The max composition scheme simply uses the
+     * viscosity of whichever field has the highest volume fraction.
+     */
+    enum CompositionalAveragingOperation
+    {
+      harmonic,
+      arithmetic,
+      geometric,
+      maximum_composition
+    };
+
+    /**
+     * For multicomponent material models:
+     * Material models compute output quantities such as the viscosity, the
+     * density, etc. For some models, these values depend strongly on the
+     * composition, and more than one compositional field might have nonzero
+     * values at a given quadrature point. This means that properties have to
+     * be averaged based on the fractions of each compositional field present.
+     * This function performs this type of averaging, and allows it to choose
```

Comment:
Replace with "The averaging is based on the choice in @p average_type." No reason to list all options here.